### PR TITLE
Change import protocol service to build protocol service [SCI-3544]

### PR DESCRIPTION
**Pull request by jbargu**
Jira ticket: [SCI-3544](https://biosistemika.atlassian.net/browse/SCI-3544)

### What was done
Change import protocol service to build protocol service and decouple
`API client` from `Normalizer`.

#### ToDo:
With this change `ApiClient`-s should inherit from the Base protocol
importer as [done here](https://github.com/biosistemika/scinote-web/pull/1795).

### CODE REVIEW

File: app/services/protocol_importers/build_protocol_from_client_service.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      @team = Team.find team_id
+      @errors = {}
+    end
+
+    def call
+      return self unless valid?
+
+      # TODO: check for errors
+      api_response = api_client.single_protocol(id: @id)
+      normalized_hash = normalizer.load_protocol(api_response.parsed_response)
```
Comment (mlorb) [inline app/services/protocol_importers/build_protocol_from_client_service.rb]: Didn't you rename `load_protocol` to `normalize_protocol`?

File: app/services/protocol_importers/build_protocol_from_client_service.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+      pio = ProtocolImporters::ProtocolIntermediateObject.new(normalized_json: normalized_hash,
+                                                              user: @user,
+                                                              team: @team)
+
```
Comment (mlorb) [inline app/services/protocol_importers/build_protocol_from_client_service.rb]: I think you are missing a line here:
`@pio_protocol = pio.build`

File: spec/services/protocol_importers/build_protocol_from_client_service_spec.rb [hunk 3]
```diff
@@ -25,13 +25,10 @@
   context 'when have valid arguments' do
     before do
       allow_any_instance_of(ProtocolImporters::ProtocolsIO::V3::ProtocolNormalizer)
-        .to(receive(:load_protocol).and_return(normalized_response))
+        .to(receive(:normalize_protocol).and_return(normalized_response))
       # Do not generate and request real images
       allow(ProtocolImporters::AttachmentsBuilder).to(receive(:generate).and_return([]))
     end
-    it 'Adds Protocol record' do
```
Comment (mlorb) [inline spec/services/protocol_importers/build_protocol_from_client_service_spec.rb]: Why did you remove this test?

Comment (jbargu) [reply spec/services/protocol_importers/build_protocol_from_client_service_spec.rb]: It didn't pass, helloo? :D This class only builds the protocol, no longer imports it to DB.

File: app/services/protocol_importers/build_protocol_from_client_service.rb (hunk 2 above)
Comment (jbargu) [reply app/services/protocol_importers/build_protocol_from_client_service.rb]: Nice catch!

File: spec/services/protocol_importers/build_protocol_from_client_service_spec.rb (hunk 3 above)
Comment (jbargu) [reply spec/services/protocol_importers/build_protocol_from_client_service_spec.rb]: I added another one to catch the above bugs.
